src/emulator/functionsEmulator.ts

```diff
@@ -459,33 +457,35 @@ export class FunctionsEmulator implements EmulatorInstance {
         `No node binary for ${emulatableBackend.functionsDir}. This should never happen.`
       );
     }
-    const worker = this.invokeRuntime(
-      this.getBaseBundle(emulatableBackend),
-      {
-        nodeBinary: emulatableBackend.nodeBinary,
-        extensionTriggers: emulatableBackend.predefinedTriggers,
-      },
-      // Don't include user envs when parsing triggers.
-      {
-        ...this.getSystemEnvs(),
-        ...this.getEmulatorEnvs(),
-        FIREBASE_CONFIG: this.getFirebaseConfig(),
-        ...emulatableBackend.env,
-      }
-    );
-
-    const triggerParseEvent = await EmulatorLog.waitForLog(
-      worker.runtime.events,
-      "SYSTEM",
-      "triggers-parsed"
-    );
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-    const parsedDefinitions = triggerParseEvent.data
-      .triggerDefinitions as ParsedTriggerDefinition[];
-
-    const triggerDefinitions: EmulatedTriggerDefinition[] =
-      emulatedFunctionsByRegion(parsedDefinitions);
 
+    let triggerDefinitions: EmulatedTriggerDefinition[];
+    if (emulatableBackend.predefinedTriggers) {
+      triggerDefinitions = emulatedFunctionsByRegion(emulatableBackend.predefinedTriggers);
+    } else {
+      const runtimeDelegate = await getRuntimeDelegate({
+        projectId: this.args.projectId,
+        projectDir: this.args.projectDir,
+        sourceDir: emulatableBackend.functionsDir,
+        runtime: "",
+      });
+      logger.debug(`Validating ${runtimeDelegate.name} source`);
+      await runtimeDelegate.validate();
+      logger.debug(`Building ${runtimeDelegate.name} source`);
+      await runtimeDelegate.build();
+      logger.debug(`Analyzing ${runtimeDelegate.name} backend spec`);
+      const discoveredBackend = await runtimeDelegate.discoverSpec(
+        {},
+        // Don't include user envs when parsing triggers.
+        {
+          ...this.getSystemEnvs(),
+          ...this.getEmulatorEnvs(),
+          FIREBASE_CONFIG: this.getFirebaseConfig(),
+          ...emulatableBackend.env,
+        }
+      );
+      const endpoints = backend.allEndpoints(discoveredBackend);
+      triggerDefinitions = emulatedFunctionsFromEndpoints(endpoints);
```

Comment:
is "emulated function" going to barf on scheduled, task queue, etc. functions?

Reply:
It's never barfed on scheduled function, though it will silently ignore task queue and other unsupported functions. The user will warning log like this one:

```
          `Trigger "${definition.name}" has neither "httpsTrigger" or "eventTrigger" member`
```

to let them know about these unknown triggers.

Reply:
TQ is such a nice project. Let's talk about it more this Friday!